feat: Add game clear reward gem

Koominjae opened the pull request:
## Add function to give 1 gem when player clears game
1 gem is given every time each game is cleared. ('Clear' means when you complete without dying at the last level)

## Changes
`ScoreScreen.java`
   - Add `saveGem()` for increasing 1 gem
   - Add conditions to draw game clear screen and load `saveGem()` only when game is cleared

`DrawManager.java`
   - Add `drawGameClear()` to draw game clear screen

![clear screen (소형)](https://github.com/user-attachments/assets/31858e40-c15a-419c-9284-57369cd8e319)
![log](https://github.com/user-attachments/assets/1afc3633-5c2f-482b-9279-37546cfa6ff0)
![clear gem (소형)](https://github.com/user-attachments/assets/d48c0d30-18e4-4689-82df-f255c7c416d1)

Koominjae committed:
feat:Add saveGem method to ScoreScreen

Update : Added a method to save gem because 1 gem is given for game clear.

ScoreScreen.java
- add new method : saveGem()

Koominjae committed:
feat: Add conditions for game clear to ScoreScreen

Update : Added conditions to update method so that 1 gem is given only when the game is cleared

ScoreScreen.java
	update()
	- add new condition : livesRemaining > 0 && level > 7

Koominjae committed:
feat: Add game clear screen

Update : Added a new screen for game clear to distinguish between game over and game clear.

ScoreScreen.java
	draw()
	- add new condition : livesRemaining > 0 && level > 7 then draw game clear screen

DrawManager.java
- add new method : drawGameClear(screen, acceptsInput, isNewRecord)

suhynnoh commented on src/engine/DrawManager.java:
```diff
@@ -544,6 +544,37 @@ public void drawNameInput(final Screen screen, final char[] name,
+	// CtrlS
+	public void drawGameClear(final Screen screen, final boolean acceptsInput,
+							 final boolean isNewRecord) {
+		String gameClearString = "Game Clear";
```
`drawGameClear` 메소드와 `drawGameOver` 메소드가 이 한 줄을 제외하곤 전부 동일한 것 같습니다.

기존 `drawGameOver` 메소드의 이름을 `drawGameEnd` 로 변경하고, `isGameClear` 라는 불리언 변수를 입력 받아 `gameEndString` 을 분기하는 건 어떨까요?

```suggestion
		String gameEndString = isGameClear ? "Game Clear" : "Game Over";
```

suhynnoh commented on src/screen/ScoreScreen.java:
```diff
@@ -240,14 +246,32 @@ private void saveCoin() {
 
-		drawManager.drawGameOver(this, this.inputDelay.checkFinished(),
-				this.isNewRecord);
+		if (this.livesRemaining > 0 && this.level > 7) {
+			drawManager.drawGameClear(this, this.inputDelay.checkFinished(),
+					this.isNewRecord);
+		} else {
+			drawManager.drawGameOver(this, this.inputDelay.checkFinished(),
+					this.isNewRecord);
+		} // CtrlS
```
this.isGameClear 필드를 추가해, ScoreScreen 생성자에서 작성해주신 조건문을 활용해 isGameClear를 초기화 할 수 있을 것 같습니다.

수정된 drawGameEnd는 다음과 같이 isGameClear 필드를 추가해서 호출합니다.
```suggestion
		drawManager.drawGameEnd(this, this.inputDelay.checkFinished(), this.isNewRecord, this.isGameClear); // CtrlS
```

Koominjae replied:
불필요한 코드 라인 수 증가도 피하고 좋은 거 같습니다. 반영해보도록 하겠습니다.

Koominjae committed:
refactor: Modify process of determining game clear

Update : Added boolean variable to determine game clear

ScoreScreen.java
- add new field : isGameClear
	update()
	- modify condition
	draw()
	- modify to load drawGameEnd() not drawGameClear(), drawGameOver()

DrawManager.java
- delete method : drawGameClear()
- modify method name : drawGameOver() to drawGameEnd()
	drawGameEnd()
	- add new parameter : boolean isGameClear
	- Depending on isGameClear, it is decided whether to draw game clear screen or game over screen

suhynnoh reviewed (approved):
빠른 반영 감사합니다. `saveGem` 호출하는 부분도 `isGameClear` 로 바꿀 수 있는 것은 생각 못했는데, 이 부분도 좋은 것 같네요. 승인했습니다!

Geguuuul commented on src/screen/ScoreScreen.java:
```diff
@@ -240,14 +249,27 @@ private void saveCoin() {
+	private void saveGem() {
+		try {
+			Core.getCurrencyManager().addGem(1);
+			logger.info("You eared 1 Gem for Game Clear");
```
You earned에서 n에 빠져 있습니다.

Geguuuul reviewed (changes requested):
log 출력 부분에 You eared를 You earned로만 고쳐주시면 될 것 같습니다.

Geguuuul commented on src/screen/ScoreScreen.java:
```diff
@@ -240,14 +249,27 @@ private void saveCoin() {
+	private void saveGem() {
+		try {
+			Core.getCurrencyManager().addGem(1);
+			logger.info("You eared 1 Gem for Game Clear");
```
```suggestion
			logger.info("You earned 1 Gem for Game Clear");
```

Koominjae replied:
수정했습니다

Geguuuul reviewed (changes requested):
You eared에서 You earned로 고치면 될 것 같습니다. 이외에는 괜찮은 것 같습니다.

Koominjae committed:
chore: correct wrong character
	modified:   src/screen/ScoreScreen.java
